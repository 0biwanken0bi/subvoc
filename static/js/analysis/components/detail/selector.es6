import $ from 'jquery';
import preact from 'preact';


class WordDetailSelector extends preact.Component {
    render({ active, enabled, code, label, onSelectPOS }) {
        const classNames = `tab ${enabled ? '' : 'empty'} ${active ? 'active' : ''}`;
        return <div onClick={() => enabled ? onSelectPOS(code) : null} class={classNames} >
            <span>{ label }</span>
        </div>
    }
}


export { WordDetailSelector }
