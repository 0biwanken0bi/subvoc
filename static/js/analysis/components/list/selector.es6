import $ from 'jquery';
import preact from 'preact';


class DifficultyGroup extends preact.Component {
    render({ level, label, count, active, onSelect }) {
        const classNames = `group card ${ active ? 'active' : '' }`;
        return <div class={classNames} onclick={() => onSelect(level)}>
            <div class="label">
                { label.toLowerCase() }
            </div>
            <div class="count badge">
                { count }
            </div>
        </div>
    }
}


class DifficultySelector extends preact.Component {

    constructor(props) {
        super(props);

        const groups = {};
        $(props.words).each(function(idx, word) {
            groups[word.difficulty.label] = {
                level: word.difficulty.level,
                count: (groups[word.difficulty.label] || { count: 0 }).count + 1,
            }
        });

        this.state = { groups };
    }

    render({ selected, onSelect, words }) {
        return <div class="difficulty">
            { $.map(Object.keys(this.state.groups),
                (label) => {
                    const group = this.state.groups[label];
                    return <DifficultyGroup
                                level={group.level}
                                count={group.count}
                                active={selected === group.level}
                                label={label.toLowerCase()}
                                onSelect={onSelect} />
                })
            }
        </div>
    }
}


export { DifficultySelector };
