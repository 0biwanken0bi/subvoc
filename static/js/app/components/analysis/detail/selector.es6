import React from 'react';
import classNames from 'classnames';


const PARTS_OF_SPEACH = ['noun', 'verb', 'adj', 'adv'];

function getFreq(word, pos) {
    return (word.byPOS[pos] || {}).freq || 0;
}

function getExcerpts(word, pos) {
    return (word.byPOS[pos] || {}).excerpts || [];
}

function getDefinitions(word, pos) {
    return word.lookup[pos] || [];
}


const WordPartOfSpeachItem = ({ active, enabled, label, freq, onSelect }) => {
    const classes = classNames(label, 'tab', 'card', {'empty': !enabled}, {active});
    return <div onClick={() => enabled ? onSelect(label) : null} className={classes} >
        <div className="label">
            { label }
        </div>
        { freq ? <div className="count badge">{ freq }</div> : <div className="count">&nbsp;</div> }
    </div>;
};


const WordPartOfSpeachSelector = ({selected, word, onSelect}) => {
    return <div>
        { PARTS_OF_SPEACH.map((pos) => 
            <WordPartOfSpeachItem
                key={pos}
                active={selected === pos}
                enabled={(getExcerpts(word, pos).length || getDefinitions(word, pos).lenght) > 0}
                label={pos}
                freq={getFreq(word, pos)}
                onSelect={onSelect} />
        ) }
    </div>;
};


export { 
    getDefinitions, 
    getExcerpts, 
    PARTS_OF_SPEACH, 
    WordPartOfSpeachItem,
    WordPartOfSpeachSelector
};
