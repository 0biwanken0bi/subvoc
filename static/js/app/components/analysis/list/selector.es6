import React from 'react';


const DifficultyGroup = ({ level, label, count, active, onSelect }) => {
    const classNames = `group card ${ active ? 'active' : '' }`;
    return <div className={classNames} onClick={() => onSelect(level)}>
        <div className="label">
            { label.toLowerCase() }
        </div>
        <div className="count badge">
            { count }
        </div>
    </div>;
};


const DifficultySelector = ({ selected, onSelect, words }) => {
    const groups = {};
    words.forEach((word) => {
        groups[word.difficulty.label] = {
            level: word.difficulty.level,
            count: (groups[word.difficulty.label] || { count: 0 }).count + 1,
        };
    });

    return <div className="difficulty">
        { Object.keys(groups).map((label) => {
            const group = groups[label];
            return <DifficultyGroup
                        key={group.level}
                        level={group.level}
                        count={group.count}
                        active={selected === group.level}
                        label={label.toLowerCase()}
                        onSelect={onSelect} />;
        }) }
    </div>;
};


export { DifficultySelector };
