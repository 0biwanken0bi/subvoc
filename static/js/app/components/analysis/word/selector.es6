import React from 'react';
import classNames from 'classnames';


const PARTS_OF_SPEECH = {
    'noun': 'noun',
    'verb': 'verb',
    'adjective': 'adj',
    'adverb': 'adv'
}


function getFreq(word, pos) {
    return (word.byPOS[pos] || {}).freq || 0;
}

function getExcerpts(word, pos) {
    return (word.byPOS[pos] || {}).excerpts || [];
}

function getDefinitions(word, pos) {
    return word.lookup[pos] || [];
}


const PartOfSpeechItem = ({ type, enabled, active, label, freq, onSelect }) => {
    const classes = classNames(label, 'tab', 'card', {'empty': !enabled}, {active});
    return <div onClick={() => enabled ? onSelect(type) : null} className={classes} >
        <div className="label">
            { label }
        </div>
        { freq ? <div className="count badge">{ freq }</div> : <div className="count">&nbsp;</div> }
    </div>;
};


const PartOfSpeechSelector = ({selected, word, onSelect}) => 
    <div>
        { Object.keys(PARTS_OF_SPEECH).map(pos =>
            <PartOfSpeechItem
                key={pos}
                type={pos}
                enabled={(getExcerpts(word, pos).length || getDefinitions(word, pos).length) > 0}
                active={selected === pos}
                label={PARTS_OF_SPEECH[pos]}
                freq={getFreq(word, pos)}
                onSelect={onSelect} />
        ) }
    </div>;


export { 
    getDefinitions, 
    getExcerpts, 
    PARTS_OF_SPEECH, 
    PartOfSpeechItem,
    PartOfSpeechSelector
};
