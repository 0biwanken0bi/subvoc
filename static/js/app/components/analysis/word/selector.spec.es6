import React from 'react';

import { shallow, render } from 'enzyme';
import td from 'testdouble';

import { PartOfSpeachItem, PartOfSpeachSelector } from './selector.es6';


describe("<PartOfSpeachSelector />", () => {

    test("should render", () => {
        const word = { byPOS: {}, lookup: {} };
        const result = shallow(
            <PartOfSpeachSelector word={word} />
        );
        
        expect(result).toMatchSnapshot();
    });

    test("should mark selected POS as 'active'", () => {
        const word = { byPOS: {}, lookup: {} };
        const result = shallow(
            <PartOfSpeachSelector selected='noun' word={word} />
        );
        
        expect(result).toMatchSnapshot();
    });

    test("should mark POS with excerpt as available", () => {
        const word = {
            byPOS: { 
                noun: { freq: 1, excerpts: ['...'] },
                verb: { freq: 1, excerpts: ['...'] }
            },
            lookup: {}
        };

        const result = shallow(
            <PartOfSpeachSelector selected='noun' word={word} />
        );
        
        expect(result).toMatchSnapshot();
    });

    test("should mark POS with definition as available", () => {
        const word = {
            byPOS: { 
                noun: { freq: 1, excerpts: ['...'] }
            }, 
            lookup: {
                verb: ['...']
            }
        };

        const result = shallow(
            <PartOfSpeachSelector word={word} />
        );

        expect(result).toMatchSnapshot();
    });
});

describe("<PartOfSpeachItem />", () => {
    let onSelect;

    beforeEach(() => {
        onSelect = td.function();
    });

    test("should render disabled, inactive POS item", () => {
        const result = render(
            <PartOfSpeachItem enabled={false} label='noun' />
        );

        expect(result).toMatchSnapshot();
    });

    test("should mark POS item as enabled", () => {
        const result = render(
            <PartOfSpeachItem enabled={true} label='noun' />
        );

        expect(result).toMatchSnapshot();
    });

    test("should mark POS item as active", () => {
        const result = render(
            <PartOfSpeachItem enabled={true} active={true} label='noun' />
        );

        expect(result).toMatchSnapshot();
    });

    test("should mark POS item with frequency", () => {
        const result = render(
            <PartOfSpeachItem enabled={true} freq={1} label='noun' />
        );

        expect(result).toMatchSnapshot();
    });

    test("should propagate click on enabled POS item", () => {
        const result = shallow(
            <PartOfSpeachItem enabled={true} label='noun' onSelect={onSelect} />
        );

        result.find('.tab.noun').simulate('click');
        td.verify(onSelect('noun'));
    });

    test("should not propagate click on disabled POS item", () => {
        const result = shallow(
            <PartOfSpeachItem enabled={true} label='noun' onSelect={onSelect} />
        );

        result.find('.tab.noun').simulate('click');
        td.verify(onSelect, { times: 0 });
    });
});