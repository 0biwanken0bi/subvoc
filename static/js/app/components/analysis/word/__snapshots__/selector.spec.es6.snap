// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PartOfSpeechItem /> should mark POS item as active 1`] = `
<div
  class="noun tab card active"
>
  <div
    class="label"
  >
    noun
  </div>
  <div
    class="count"
  >
     
  </div>
</div>
`;

exports[`<PartOfSpeechItem /> should mark POS item as enabled 1`] = `
<div
  class="noun tab card"
>
  <div
    class="label"
  >
    noun
  </div>
  <div
    class="count"
  >
     
  </div>
</div>
`;

exports[`<PartOfSpeechItem /> should mark POS item with frequency 1`] = `
<div
  class="noun tab card"
>
  <div
    class="label"
  >
    noun
  </div>
  <div
    class="count badge"
  >
    1
  </div>
</div>
`;

exports[`<PartOfSpeechItem /> should render disabled, inactive POS item 1`] = `
<div
  class="noun tab card empty"
>
  <div
    class="label"
  >
    noun
  </div>
  <div
    class="count"
  >
     
  </div>
</div>
`;

exports[`<PartOfSpeechSelector /> should mark POS with definition only as available 1`] = `
<div>
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="noun"
    type="noun"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="verb"
    type="verb"
  />
  <PartOfSpeechItem
    active={false}
    enabled={true}
    freq={0}
    label="adj"
    type="adjective"
  />
  <PartOfSpeechItem
    active={true}
    enabled={true}
    freq={1}
    label="adv"
    type="adverb"
  />
</div>
`;

exports[`<PartOfSpeechSelector /> should mark POS with excerpt as active 1`] = `
<div>
  <PartOfSpeechItem
    active={false}
    enabled={true}
    freq={0}
    label="noun"
    type="noun"
  />
  <PartOfSpeechItem
    active={false}
    enabled={true}
    freq={1}
    label="verb"
    type="verb"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adj"
    type="adjective"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adv"
    type="adverb"
  />
</div>
`;

exports[`<PartOfSpeechSelector /> should mark POS with excerpt as available 1`] = `
<div>
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="noun"
    type="noun"
  />
  <PartOfSpeechItem
    active={false}
    enabled={true}
    freq={1}
    label="verb"
    type="verb"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adj"
    type="adjective"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adv"
    type="adverb"
  />
</div>
`;

exports[`<PartOfSpeechSelector /> should mark selected POS as 'active' 1`] = `
<div>
  <PartOfSpeechItem
    active={true}
    enabled={false}
    freq={0}
    label="noun"
    type="noun"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="verb"
    type="verb"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adj"
    type="adjective"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adv"
    type="adverb"
  />
</div>
`;

exports[`<PartOfSpeechSelector /> should render 1`] = `
<div>
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="noun"
    type="noun"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="verb"
    type="verb"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adj"
    type="adjective"
  />
  <PartOfSpeechItem
    active={false}
    enabled={false}
    freq={0}
    label="adv"
    type="adverb"
  />
</div>
`;
